instances.js

```diff
@@ -0,0 +1,137 @@
+/*!
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+import axios from 'axios';
+
+const route = '/vc-issuer/instances';
+
+export async function create({instance}) {
+  const url = route;
+  const response = await axios.post(url, instance);
+  return response.data;
+}
+
+export async function setIssuer({id, controller, presentation}) {
+  const url = `${route}/${encodeURIComponent(id)}/issuer`;
+  const response = await axios.post(url, {controller, presentation});
+  return response.data;
+}
+
+export async function get({id}) {
+  const url = `${route}/${encodeURIComponent(id)}`;
+  const response = await axios.get(url);
+  return response.data;
+}
+
+export async function getAll({controller} = {}) {
+  const instanceIds = await _getInstanceIds({id: controller});
+  const promises = instanceIds.map(async id => {
+    try {
+      const instance = await get({id});
+      return instance;
+    } catch(e) {
+      console.log(`Unable to fetch issuer: "${id}"`);
+      console.error(e);
+    }
+  });
+  // Resolve promises and filter out undefined
+  // FIXME: Consider pulling in promises lib to limit concurrency
+  const instances = await Promise.all(promises);
+  return instances.filter(promise => promise);
+}
+
+export async function remove({id}) {
+  const url = `${route}/${encodeURIComponent(id)}`;
+  try {
+    await axios.delete(url);
+    return true;
+  } catch(e) {
+    if(e.name === 'NotFoundError') {
+      return false;
+    }
+    throw e;
+  }
+}
+
+export async function _getInstanceIds({id}) {
+  const zcaps = await getCapabilities({id});
+  const ids = [];
+  zcaps.forEach(({referenceId}) => {
+    // assumes instance ids are 36 characters and prepended on referenceId
+    const instanceId = referenceId.slice(0, 36);
```

Comment:
This looks fragile.

Reply:
Agreed. What is your recommendation for a proposed change? Currently, the Issuer Instance's ID is a uuid. That uuid is prepended to other data to create the reference id. I could not think of another way of getting the information I need.